### PR TITLE
Apply type annotations to x509 ct and ocsp

### PR DESCRIPTION
**Pull request by alex**
Refs #5670

### CODE REVIEW

File: src/cryptography/fernet.py [hunk 1]
```diff
@@ -157,8 +157,8 @@ def _decrypt_data(
 
 
 class MultiFernet(object):
-    def __init__(self, fernets_it: typing.Iterator[Fernet]):
-        fernets = list(fernets_it)
+    def __init__(self, fernets: typing.Iterable[Fernet]):
```
Comment (reaperhulk) [inline src/cryptography/fernet.py:160]: Changing this back to match the docs?

Comment (alex) [reply src/cryptography/fernet.py:160]: `list` isn't iterator, but it is iterable

**Review by reaperhulk — CHANGES_REQUESTED** [review]
`_OCSPRequest` and `_OCSPResponse`, as well as `OCSPRequest` and `OCSPResponse` (the ABCs) are missing type information on quite a few properties. `Extensions` is as well.

`RFC822Name`, `DNSName`, `UniformResourceIdentifier`, `DirectoryName`, `RegisteredID`, `IPAddress`, `OtherName`,  `NameAttribute`, and more don't have typing on various dunder methods, while we *did* add typing to those methods in past PRs (e.g. https://github.com/pyca/cryptography/pull/5711/files#diff-c42cf4fc4d5c5eacf9b7cda43dbb0ab06335d0b9804814a20f488a827987d847R48).

Maybe we should just strip the typing on dunder methods?

File: src/cryptography/x509/ocsp.py [hunk 2]
```diff
@@ -184,279 +137,339 @@ def __init__(
...
+class OCSPResponseBuilder(object):
+    def __init__(
+        self, response=None, responder_id=None, certs=None, extensions=[]
+    ):
+        self._response = response
+        self._responder_id = responder_id
+        self._certs = certs
+        self._extensions = extensions
+
+    def add_response(
```
Comment (reaperhulk) [inline src/cryptography/x509/ocsp.py:359]: All these functions should have `-> "OCSPResponseBuilder"` for return type so mypy can do chaining.

File: src/cryptography/x509/ocsp.py [hunk 3]
```diff
@@ -184,279 +137,339 @@ def __init__(
...
         The list of single response extensions. Not response extensions.
         """
+
+
+class OCSPRequestBuilder(object):
+    def __init__(self, request=None, extensions=[]):
+        self._request = request
+        self._extensions = extensions
+
+    def add_certificate(
```
Comment (reaperhulk) [inline src/cryptography/x509/ocsp.py:313]: All these functions should have `-> "OCSPRequestBuilder"` for return type so mypy can do chaining.

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**